Engine/Component/ComponentMeshRenderer.cpp

```diff
@@ -65,14 +64,28 @@ void ComponentMeshRenderer::Render()
 	glUseProgram(program);
 
 
-	glUniformMatrix4fv(glGetUniformLocation(program, "palette"), palette.size(), GL_TRUE, &palette[0][0][0]);
-	glUniform1i(glGetUniformLocation(program, "num_joints"), skeleton_uuid != 0 ? MAX_JOINTS : 1);
+	if (palette.size() > 0)
+	{
+		glUniformMatrix4fv(glGetUniformLocation(program, "palette"), palette.size(), GL_TRUE, &palette[0][0][0]);
+	}
+	glUniform1i(glGetUniformLocation(program, "has_skinning_value"), skeleton_uuid != 0 ? 0 : 1);
 
 	glBindBuffer(GL_UNIFORM_BUFFER, App->program->uniform_buffer.ubo);
 	glBufferSubData(GL_UNIFORM_BUFFER, App->program->uniform_buffer.MATRICES_UNIFORMS_OFFSET, sizeof(float4x4), owner->transform.GetGlobalModelMatrix().Transposed().ptr());
 	glBindBuffer(GL_UNIFORM_BUFFER, 0);
 
 	App->lights->Render(owner->transform.GetGlobalTranslation(), program);
+
+	if (material_to_render->GetMaterialTexture(Material::MaterialTextureType::SPECULAR) != nullptr)
+		material_to_render->use_specular_map = true;
+	else
+		material_to_render->use_specular_map = false;
+
+	if (material_to_render->GetMaterialTexture(Material::MaterialTextureType::NORMAL) != nullptr)
+		material_to_render->use_normal_map = true;
+	else
+		material_to_render->use_normal_map = false;
```

Comment:
This can be simplified to
```
material_to_render->use_specular_map = material_to_render->GetMaterialTexture(Material::MaterialTextureType::SPECULAR) != nullptr;
material_to_render->use_normal_map = material_to_render->GetMaterialTexture(Material::MaterialTextureType::NORMAL) != nullptr
```